packages/content-management/favorites/favorites_server/src/favorites_routes.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreSetup, Logger, SavedObjectsClient } from '@kbn/core/server';
+import { once } from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { FavoritesService } from './favorites_service';
+import { favoritesSavedObjectType } from './favorites_saved_object';
+
+// only dashboard is supported for now
+// TODO: make configurable or allow any string
+const typeSchema = schema.oneOf([schema.literal('dashboard')]);
+
+/**
+ * @public
+ * Response for get favorites API
+ */
+export interface GetFavoritesResponse {
+  favoriteIds: string[];
+}
+
+export function registerFavoritesRoutes({ core, logger }: { core: CoreSetup; logger: Logger }) {
+  const router = core.http.createRouter();
+
+  const getSavedObjectClient = once(async () => {
+    const [coreStart] = await core.getStartServices();
+
+    // We need an internal client to access the `favorite` type which not every user has access to
+    // and give access only to the current user's favorites through this API
+    const internalClient = new SavedObjectsClient(
+      coreStart.savedObjects.createInternalRepository([favoritesSavedObjectType.name])
+    );
+    return internalClient;
+  });
+
+  router.post(
```

Comment:
__nit:__ Would you mind adding a comment in the code for these routes explaining why we don't need to protect them with any access tag? We started recommending this recently to make it easier to audit HTTP routes that aren't tied to any specific privileges and can be called by any user. Soon, we'll introduce a dedicated API for routes to make providing reasoning for opting-out of route security easier.